cmd/samples/common/factory.go

```diff
@@ -17,10 +20,30 @@ import (
 )
 
 const (
-	_cadenceClientName      = "cadence-client"
-	_cadenceFrontendService = "cadence-frontend"
+	_cadenceClientName            = "cadence-client"
+	_cadenceFrontendService       = "cadence-frontend"
+	_authorizationTokenHeaderName = "cadence-authorization"
 )
 
+type authOutboundMiddleware struct {
+	authProvider worker.AuthorizationProvider
+}
+
+func (m *authOutboundMiddleware) Call(ctx context.Context, request *transport.Request, out transport.UnaryOutbound) (*transport.Response, error) {
+	if m.authProvider == nil {
+		return out.Call(ctx, request)
+	}
+
+	token, err := m.authProvider.GetAuthToken()
+	if err != nil {
+		return nil, err
+	}
+	request.Headers = request.Headers.
+		With(_authorizationTokenHeaderName, string(token))
```

Comment:
<img width="663" alt="Screenshot 2023-08-11 at 5 00 41 PM" src="https://github.com/uber-common/cadence-samples/assets/2725764/3fefc3a5-4f94-4ae5-9a71-4b7761449b0c">
Should this be `Authorization` as the header key?

Reply:
I'm not sure if there's a clear standard around this, but I seem to recall there was a semi-vibes based agreement that was passed around various systems with that header and the prefix `Bearer`, at least for the systems I'd worked on previously

Reply:
I'm checking more things related to jwt/oauth, so yeah, will come back later on this one. Closing it for now